src/main/webapp/script.js

```diff
@@ -117,59 +117,120 @@ function getLoginStatus() {
 function hardCodedRecipeCard() {
   const displayRecipeElement = document.getElementById('display-recipes');
   displayRecipeElement.innerHTML = "";
-  displayRecipeElement.appendChild(createRecipeElement());
+  // recipes.forEach(recipe => displayRecipeElement.appendChild(createRecipeElement(recipe)));
+  displayRecipeElement.appendChild(createRecipeElement("Placeholder for recipe json"));
 }
 
-/** Creates an element that represents a recipe */
+/** Creates an element that represents a recipe card */
 // currently only has hardcoded values
-function createRecipeElement() {
+function createRecipeElement(recipe) {
   const recipeElement = document.createElement('div');
   recipeElement.className = 'recipe-card';
 
   const titleElement = document.createElement('p');
-  titleElement.innerText = "Title";
+  titleElement.innerText = "Title"; // replace with recipe title
+  titleElement.className = "recipe-card-title";
   recipeElement.appendChild(titleElement);
 
+  const recipeTableElement = document.createElement('p');
+  recipeTableElement.className = "recipe-card-table";
+
   const infoElements = document.createElement('div');
   infoElements.className = 'recipe-card-block';
 
   const imageElement = document.createElement('img');
-  imageElement.src = "/images/salad.jpeg";
+  imageElement.src = "/images/salad.jpeg"; // replace with recipe image url
   infoElements.appendChild(imageElement);
 
   const linkElement = document.createElement('a');
-  linkElement.href = "https://www.w3schools.com/jsref/dom_obj_image.asp";
+  linkElement.href = "https://www.w3schools.com/jsref/dom_obj_image.asp"; // replace with recipe instructions url
   linkElement.innerHTML = "<br/>random link";
   infoElements.appendChild(linkElement);
 
-  recipeElement.appendChild(infoElements);
+  recipeTableElement.appendChild(infoElements);
+
+  recipeTableElement.appendChild(createRecipeCardAlerts(recipe));
+  recipeElement.append(recipeTableElement);
+  
+  const tagElements = createRecipeCardTags(recipe);
+  recipeElement.append(tagElements);
+
+  const tagTableElement = document.createElement('p');
+  tagTableElement.className = "recipe-card-table";
 
+  const tagTextElement = document.createElement('textarea');
+  tagTableElement.appendChild(tagTextElement);
+
+  const addTagElement = document.createElement('button');
+  addTagElement.innerText = '+';
+  addTagElement.className = "add-tag-button";
+  addTagElement.addEventListener('click', () => {
+    const newTagName = (tagTextElement.value).trim();
+    if (newTagName != "") {
+      // TODO: will eventually have post request to add tags
+      tagElements.appendChild(createTagElement(newTagName));
+    }
+  });
+  tagTableElement.append(addTagElement);
+  recipeElement.appendChild(tagTableElement);
+
+  return recipeElement;
+}
```

Comment:
Is it possible to define an string template for the html content, and replace some variables using the properties from the recipe object?

I'm just wondering because the current way make me a bit difficult to understand the overall page layout. It doesn't need to be resolved immediately.

Reply:
That is a good idea, I will try to figure that out.